gui/src/components/Taipy/Chart.tsx

```diff
@@ -212,12 +209,16 @@ const Chart = (props: ChartProp) => {
     useEffect(() => {
         if (refresh || !data[dataKey.current]) {
             const backCols = Object.keys(config.columns).map((col) => config.columns[col].dfid);
+            const xAxis = layout.xaxis.title instanceof Object && "text" in layout.xaxis.title ? layout.xaxis.title.text : layout.xaxis.title;
```

Comment:
Why not use 
```
const xAxis = config.traces.length && config.traces[0].length && config.traces[0][0] && config.columns[config.traces[0][0]].dfid;
const yAxis = config.traces.length == 1 && config.traces[0].length > 1 && config.columns[config.traces[0][1]] &&  config.columns[config.traces[0][1]].dfid;
```
That would give you the real column names (and not the titles associated to the axis)

Reply:
And we need to update the dependency list with config.columns & config.traces